python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/abstract_interp_lib_gen.py

```diff
@@ -605,6 +605,27 @@ def adaptive_avg_pool1d(self: List[int], out: List[int]):
 
     return shape
 
+
+# TODO: This should be upstreamed.
+# See https://github.com/pytorch/pytorch/pull/76889 for an example.
+def unflatten(self: List[int], dim: int, sizes: List[int]):
```

Comment:
This function has non-trivial logic. Can you add tests to it? See for an example: 

https://github.com/llvm/torch-mlir/blob/a520d39f84f159838cc2f4f88a08c9cd611635fd/python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/abstract_interp_lib_gen.py#L469-L472

Also, no need to refactor into a helper function, since the shape function `aten〇unflatten〇int〡shape`mis the only one using this.